pkg/minikube/download/iso.go

```diff
@@ -77,7 +78,10 @@ func localISOPath(u *url.URL) string {
 		return u.String()
 	}
 
-	return filepath.Join(detect.ISOCacheDir(), path.Base(u.Path))
+	urlHash := sha1.Sum([]byte(u.String()))
+	fileName := fmt.Sprintf("%s-%040x", path.Base(u.Path), urlHash)
```

Comment:
This makes the iso file name a little bit ugly to support questionable API of a single application. The common way to publish downloads is to provide a URL ending with the filename, so the current behavior makes sense.

Since the image name always end with `.iso` or `.ISO`, and the file name is likely to be in the URL even in gitlab, can be search back the URL components and take the first component ending with `.(iso|ISO)` as the file name?

Or, maybe apply this behavior only if the URL does not end with `.(iso|ISO)`? This way this change helps gitlab users, without affecting other users.

Also did you open a bug for gitlab to provide a better URL? This /download URL will cause trouble for other tools, for example what do you get with `curl -O`?

Reply:
Another way to solve this, add --iso-filename option. When set, using --iso-url will store the iso using the filename specified in --iso-filename. With this if you download the iso from a system that does not provide the common url with meaningful file name at the end, you can control the stored file name.